include/mprio.h

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2018, Henry Corrigan-Gibbs
+ * 
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef __PRIO_H__
+#define __PRIO_H__
+
+#include <msgpack.h>
+#include <nss/blapit.h>
+#include <nss/pk11pub.h>
+#include <nss/seccomon.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Seed for a pseudo-random generator (PRG). */
+#define PRG_SEED_LENGTH AES_128_KEY_LENGTH
+typedef unsigned char PrioPRGSeed[PRG_SEED_LENGTH];
+
+/*
+ * Type for each of the two servers.
+ */
+typedef enum {
+  PRIO_SERVER_A,
+  PRIO_SERVER_B
+} PrioServerId;
+
+/*
+ * Opaque types
+ */
+typedef struct prio_config *PrioConfig;
+typedef const struct prio_config *const_PrioConfig;
```

Comment:
Not sure I'm a fan of typedefing the `const` version of the struct. But nothing wrong with it I guess.

Reply:
Ah, got it. For my future reference, what is the right way to do this? Would it be better to not use these `typedef`d struct at all?

Reply:
I'd probably just use a `const *PrioConfig` etc. But not really a big deal.

Reply:
Got it. Thanks.